unity/Assets/Scripts/BaseFPSAgentController.cs

```diff
@@ -372,23 +382,38 @@ protected void HideAllAgentRenderers()
                                 break;
                             }
                         }
+
+                        //If CapsuleCast check failed, move on to next direction from p (otherwise Check 2's error message could return a false-positive)
+                        if(shouldEnqueue == false) {
+                            continue;
+                        }
+
+                        //Check 2: Is p + d within the bounds of the scene?
                         bool inBounds = agentManager.SceneBounds.Contains(newPosition);
+
                         if (errorMessage == "" && !inBounds) {
                             errorMessage = "In " +
                                 UnityEngine.SceneManagement.SceneManager.GetActiveScene().name +
                                 ", position " + newPosition.ToString() +
                                 " can be reached via capsule cast but is beyond the scene bounds.";
+                            
+                            shouldEnqueue = false;
+                            continue;
                         }
 
-                        shouldEnqueue = shouldEnqueue && inBounds && (
+                        //Check 3: Can the carried object fit into p + d's four cardinal orientations?
+                        shouldEnqueue = (
                             handObjectCanFitInPosition(newPosition, 0.0f) ||
                             handObjectCanFitInPosition(newPosition, 90.0f) ||
                             handObjectCanFitInPosition(newPosition, 180.0f) ||
                             handObjectCanFitInPosition(newPosition, 270.0f)
                         );
```

Comment:
If `shouldEnqueue = false` is set here, then `shouldEnqueue = (...` is set in the next block, setting `shouldEnqueue = false` wouldn't have ended up doing anything, since its value is being immediately overwritten. Perhaps you're missing another

```c#
if (!shouldEnqueue) {
    continue;
}
```
statement.

Reply:
No, that's not necessary either, since `shouldEnqueue` is redeclared and reinitialized after every loop iteration anyways. I kept it for clarity, but I suppose I ought to remove it.